domain/crossmodelrelation/state/model/remoteapplication_test.go

```diff
@@ -846,41 +682,58 @@ func (s *modelRemoteApplicationSuite) TestGetApplicationUUIDByOfferUUIDNotExists
 }
 
 func (s *modelRemoteApplicationSuite) TestGetRemoteApplicationConsumersSingle(c *tc.C) {
+	charmUUID := tc.Must(c, internaluuid.NewUUID).String()
 	offerUUID := tc.Must(c, internaluuid.NewUUID).String()
 	relationUUID := tc.Must(c, internaluuid.NewUUID).String()
-	appUUID := tc.Must(c, internaluuid.NewUUID).String()
-	charmUUID := tc.Must(c, internaluuid.NewUUID).String()
+	consumerModelUUID := tc.Must(c, internaluuid.NewUUID).String()
+	consumerApplicationUUID := tc.Must(c, internaluuid.NewUUID).String()
+	synthApplicationUUID := tc.Must(c, internaluuid.NewUUID).String()
 
-	// Local (offerer-side) application setup.
-	localAppUUID := tc.Must(c, internaluuid.NewUUID).String()
-	localCharmUUID := tc.Must(c, internaluuid.NewUUID).String()
+	// Offer resources needed:
+	offerApplicationUUID := tc.Must(c, internaluuid.NewUUID).String()
+	offerCharmUUID := tc.Must(c, internaluuid.NewUUID).String()
+	// Create an offer in the database.
 	s.createOffer(c, offerUUID)
-	s.createCharm(c, localCharmUUID)
-	s.createApplication(c, localAppUUID, localCharmUUID, offerUUID)
+	// Create a charm in the database.
+	s.createCharm(c, offerCharmUUID)
+	// Create an application in the database.
+	s.createApplication(c, offerApplicationUUID, offerCharmUUID, offerUUID)
 
-	ch := charm.Charm{
-		ReferenceName: "rem-1",
+	charm := charm.Charm{
+		ReferenceName: "bar",
 		Source:        charm.CMRSource,
 		Metadata: charm.Metadata{
 			Name:        "foo",
 			Description: "remote consumer application",
 			Provides: map[string]charm.Relation{
-				"db": {Name: "db", Role: charm.RoleProvider, Interface: "db", Limit: 1, Scope: charm.ScopeGlobal},
+				"db": {
+					Name:      "db",
+					Role:      charm.RoleProvider,
+					Interface: "db",
+					Limit:     1,
+					Scope:     charm.ScopeGlobal,
+				},
 			},
-			Requires: map[string]charm.Relation{},
-			Peers:    map[string]charm.Relation{},
+			Requires: map[string]charm.Relation{
+				"cache": {
+					Name:      "cache",
+					Role:      charm.RoleRequirer,
+					Interface: "cacher",
+					Scope:     charm.ScopeGlobal,
+				},
+			},
+			Peers: map[string]charm.Relation{},
```

Comment:
todo: Use only one relation endpoint in the synth charm. A synthetic charm can have one and only one relation. The params struct across the wire has a single charm relation